File: LENGUAJES/JAVASCRIPT/Ejercicio7.js
//Funciones

function obtenerNombre() {
    while (true) {
        let nombre = prompt("Ingrese su nombre:").trim();
        if (nombre) {
            return nombre;
        } else {
            alert("El nombre no puede estar vacío. Inténtalo de nuevo.");
        }
    }
}

function saludo(nombre) {
    console.log(`Hola ${nombre}, bienvenido/a!`);
}

function main() {
    while (true) {
        let nombre = obtenerNombre();
        saludo(nombre);
        
        let continuar = prompt("¿Deseas saludar a otra persona? (s/n):").trim().toLowerCase();
        if (continuar !== 's') {
            console.log("Gracias por usar el programa. ¡Hasta luego!");
            break;
        }
    }
}
main();

File: LENGUAJES/JAVA/Ejercicio7.java
//FUNCIONES
import java.util.Scanner;

public class Ejercicio7 {
    public static String obtenerNombre() {
        Scanner scanner = new Scanner(System.in);
        String nombre;

        while (true) {
            System.out.print("Ingrese su nombre: ");
            nombre = scanner.nextLine().trim();
            if (!nombre.isEmpty()) {
                return nombre;
            } else {
                System.out.println("El nombre no puede estar vacío. Inténtalo de nuevo.");
            }
        }
    }

    public static void saludo(String nombre) {
        System.out.println("Hola " + nombre + ", bienvenido/a!");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            String nombre = obtenerNombre();
            saludo(nombre);
            
            System.out.print("¿Deseas saludar a otra persona? (s/n): ");
            String continuar = scanner.nextLine().trim().toLowerCase();
            
            if (!continuar.equals("s")) {
                System.out.println("Gracias por usar el programa. ¡Hasta luego!");
                break;
            }
        }
        scanner.close(); 
    }
}